plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkContextFunction.java

```diff
@@ -57,15 +57,14 @@ static void sdkPathChanged(String newPath) {
 
   @Override
   public Object compute(IEclipseContext context, String contextKey) {
-    referencedContexts.add(context);
-
     Object path = context.get(PreferenceConstants.CLOUDSDK_PATH);
-    File location = toFile(path);
-    CloudSdk.Builder builder = new CloudSdkProvider(null).createBuilder(location);
-    if (builder == null) {
-      return NOT_A_VALUE;
+    if (path == null) {
+      // record this context as using the preference value
+      referencedContexts.add(context);
     }
-    CloudSdk instance = builder.build();
+    
+    Path location = toFile(path).toPath();
+    CloudSdk instance = new CloudSdkProvider(location).getCloudSdk();
```

Comment:
Line 61 assumes that `path` can be null. If it is, then `toFile()` will return null, leading to an NPE. So, looks like either the assumption is wrong, or we need to check null.